forc/src/ops/forc_init.rs

```diff
@@ -1,97 +1,65 @@
+    let read_the_docs = format!(
+        "Read the Docs:\n- {}\n- {}\n- {}",
+        "Sway Book: https://fuellabs.github.io/sway/latest",
+        "Rust SDK Book: https://fuellabs.github.io/fuels-rs/latest",
+        "TypeScript SDK: https://github.com/FuelLabs/fuels-ts"
+    );
+
+    let join_the_community = format!(
+        "Join the Community:\n- Follow us {}
+- Ask questions in dev-chat on {}",
+        "@SwayLang: https://twitter.com/SwayLang", "Discord: https://discord.com/invite/xfpK4Pe"
+    );
+
+    let report_bugs = format!(
+        "Report Bugs:\n- {}",
+        "Sway Issues: https://github.com/FuelLabs/sway/issues/new"
+    );
+
+    let try_forc = "To compile, use `forc build`, and to run tests use `forc test`";
+
+    info!(
+        "\n{}\n\n----\n\n{}\n\n{}\n\n{}\n\n",
+        try_forc, read_the_docs, join_the_community, report_bugs
+    );
 }
 
-#[allow(dead_code)]
-#[derive(serde::Deserialize)]
-struct GithubRepoResponse {
-    sha: String,
-    url: String,
-    // We only care about the tree here
-    tree: Vec<GithubTree>,
-    truncated: bool,
+pub fn canonicalize(path: &PathBuf) -> Result<String> {
+    match std::fs::canonicalize(path) {
+        Ok(p) => {
+            let as_str = String::from(p.to_string_lossy());
+            Ok(as_str)
+        }
+        Err(e) => anyhow::bail!("Could not derive canonical path: '{}'", e),
+    }
 }
 
-#[allow(dead_code)]
-#[derive(serde::Deserialize)]
-struct GithubTree {
-    mode: String,
-    // We only care about the "path" which are files / directory names
-    path: String,
-    sha: String,
-    size: Option<usize>,
-    #[serde(rename = "type")]
-    data_type: String,
-    url: String,
-}
+pub fn init(command: InitCommand) -> Result<()> {
+    let project_dir = match &command.path {
+        Some(p) => PathBuf::from(p),
+        None => std::env::current_dir().unwrap(),
+    };
 
-fn print_welcome_message() {
-    print_light_green("To compile, use `forc build`, and to run tests use `forc test`\n\n");
-
-    print_light_blue("Read the Docs:\n");
-    print_light_green("- Sway Book: ");
-    print_light_blue("https://fuellabs.github.io/sway/latest\n");
-    print_light_green("- Rust SDK Book: ");
-    print_light_blue("https://fuellabs.github.io/fuels-rs/latest\n");
-    print_light_green("- TypeScript SDK: ");
-    print_light_blue("https://github.com/FuelLabs/fuels-ts\n\n");
-
-    print_light_blue("Join the Community:\n");
-    print_light_green("- Follow us @SwayLang: ");
-    print_light_blue("https://twitter.com/SwayLang\n");
-    print_light_green("- Ask questions in dev-chat on Discord: ");
-    print_light_blue("https://discord.com/invite/xfpK4Pe\n\n");
-
-    print_light_blue("Report Bugs:\n");
-    print_light_green("- Sway Issues: ");
-    print_light_blue("https://github.com/FuelLabs/sway/issues/new\n");
-}
+    let project_name = command.name.unwrap_or_else(|| {
+        project_dir
+            .to_str()
+            .unwrap()
+            .split('/')
```

Comment:
This assumes Unix paths, which we should avoid whenever trivial to do so (particularly for #1526).

To retrieve the project name in the case one was not given, we should instead do something like:

```rust
project_dir.file_stem().context("failed to infer project name from directory name")?
```

https://doc.rust-lang.org/std/path/struct.Path.html#method.file_stem

Reply:
`.file_stem()` Nice 👍